tasks/eth/006-MCP-L1/SignFromJsonWithAssertions.s.sol

```diff
@@ -0,0 +1,296 @@
+    function checkOptimismMintableERC20Factory() internal view {
+        console.log("Running chain assertions on the OptimismMintableERC20Factory");
+
+        require(proxies.OptimismMintableERC20Factory.code.length != 0, "4800");
+
+        OptimismMintableERC20Factory factory = OptimismMintableERC20Factory(proxies.OptimismMintableERC20Factory);
+
+        require(factory.BRIDGE() == proxies.L1StandardBridge, "4900");
+        require(factory.bridge() == proxies.L1StandardBridge, "5000");
+    }
+
+    /// @notice Asserts that the L1ERC721Bridge is setup correctly
+    function checkL1ERC721Bridge() internal view {
+        console.log("Running chain assertions on the L1ERC721Bridge");
+
+        require(proxies.L1ERC721Bridge.code.length != 0, "5100");
+
+        L1ERC721Bridge bridge = L1ERC721Bridge(proxies.L1ERC721Bridge);
+
+        require(address(bridge.OTHER_BRIDGE()) == Predeploys.L2_ERC721_BRIDGE, "5200");
+        require(address(bridge.otherBridge()) == Predeploys.L2_ERC721_BRIDGE, "5300");
+
+        require(address(bridge.MESSENGER()) == proxies.L1CrossDomainMessenger, "5400");
+        require(address(bridge.messenger()) == proxies.L1CrossDomainMessenger, "5500");
+        require(address(bridge.superchainConfig()) == proxies.SuperchainConfig, "5600");
+    }
+
+    /// @notice Asserts the OptimismPortal is setup correctly
+    function checkOptimismPortal() internal view {
+        console.log("Running chain assertions on the OptimismPortal");
+
+        require(proxies.OptimismPortal.code.length != 0, "5700");
+
+        OptimismPortal portal = OptimismPortal(payable(proxies.OptimismPortal));
+
+        address guardian = cfg.superchainConfigGuardian();
+
+        require(address(portal.L2_ORACLE()) == proxies.L2OutputOracle, "5800");
+        require(address(portal.l2Oracle()) == proxies.L2OutputOracle, "5900");
+        require(address(portal.SYSTEM_CONFIG()) == proxies.SystemConfig, "6000");
+        require(address(portal.systemConfig()) == proxies.SystemConfig, "6100");
+        require(portal.GUARDIAN() == guardian, "6200");
+        require(portal.guardian() == guardian, "6300");
+        require(address(portal.superchainConfig()) == address(proxies.SuperchainConfig), "6400");
+        require(portal.paused() == SuperchainConfig(proxies.SuperchainConfig).paused(), "6500");
+        require(portal.l2Sender() == Constants.DEFAULT_L2_SENDER, "6600");
+    }
+
+    /// @notice Asserts that the ProtocolVersions is setup correctly
+    function checkProtocolVersions() internal view {
+        console.log("Running chain assertions on the ProtocolVersions");
+
+        require(proxies.ProtocolVersions.code.length != 0, "6700");
+
+        ProtocolVersions versions = ProtocolVersions(proxies.ProtocolVersions);
+
+        require(versions.owner() == cfg.finalSystemOwner(), "6800");
+
+        string memory json;
+
+        try vm.readFile(string.concat(vm.projectRoot(), "/tasks/eth/005-protocol-versions-ecotone/input.json"))
```

Comment:
This shows that we should be using the deploy config as the source of truth whenever we do state modifications

Reply:
Sorry @tynes what do you mean here? Want to make sure we're on the same page, ref https://github.com/ethereum-optimism/superchain-ops/pull/119#discussion_r1548622313

Reply:
I think we means that we should use the required and recommended protocol version values from DeployConfig rather than from `/tasks/eth/005-protocol-versions-ecotone/input.json"`